File: src/main/java/net/atos/client/bag/util/StatusPandEnumAdapter.java
/*
 * SPDX-FileCopyrightText: 2023 Atos
 * SPDX-License-Identifier: EUPL-1.2+
 */

package net.atos.client.bag.util;

import jakarta.json.bind.adapter.JsonbAdapter;

import nl.info.client.bag.model.generated.StatusPand;

public class StatusPandEnumAdapter implements JsonbAdapter<StatusPand, String> {

    @Override
    public String adaptToJson(final StatusPand statusPand) {
        return statusPand.toString();
    }

    @Override
    public StatusPand adaptFromJson(final String json) {
        return StatusPand.fromValue(json);
    }
}

File: src/main/kotlin/nl/info/client/bag/util/StatusPandEnumAdapter.kt
/*
 * SPDX-FileCopyrightText: 2023 Atos, 2026 INFO.nl
 * SPDX-License-Identifier: EUPL-1.2+
 */
package nl.info.client.bag.util

import jakarta.json.bind.adapter.JsonbAdapter
import nl.info.client.bag.model.generated.StatusPand

class StatusPandEnumAdapter : JsonbAdapter<StatusPand, String> {
    override fun adaptToJson(statusPand: StatusPand): String = statusPand.toString()
    override fun adaptFromJson(json: String): StatusPand = StatusPand.fromValue(json)
}
